nucliadb_sdk/nucliadb_sdk/tests/test_crud.py

```diff
@@ -0,0 +1,282 @@
+
+    await knowledgebox.async_del("mykey1")
+
+    assert await knowledgebox.async_len() == 0
+
+
+def test_dict_resource(knowledgebox: KnowledgeBox):
+    assert knowledgebox.get("mykey1") is None
+
+    knowledgebox.new_vectorset("base", 2)
+
+    resource_id = knowledgebox.upload(
+        key="mykey1",
+        binary=File(data=b"asd", filename="data"),
+        text="I'm Ramon",
+        labels=["labelset/positive"],
+        entities=[Entity(type="NAME", value="Ramon", positions=[(5, 9)])],
+        vectors=[Vector(value=[1.0, 0.2], vectorset="base")],
+    )
+    resource2 = knowledgebox[resource_id]
+
+    knowledgebox["mykey2"] = resource2
+
+    assert len(knowledgebox) == 2
+
+    del knowledgebox["mykey1"]
+
+    assert len(knowledgebox) == 1
+
+    del knowledgebox["mykey2"]
+
+    assert len(knowledgebox) == 0
+
+
+def test_update_existing_resource(knowledgebox: KnowledgeBox):
+    assert knowledgebox.get("mykey1") is None
+
+    knowledgebox.new_vectorset("base", 2)
+
+    def _validate(res1, res2):
+        assert res1.id == res2.id
+        assert res1.title != res2.title
+        assert res1.summary != res2.summary
+        assert res1.data.texts["text"] != res2.data.texts["text"]
+        assert res1.usermetadata != res2.usermetadata
+        assert res1.fieldmetadata != res2.fieldmetadata
+
+        assert res2.title == "common man"
+        assert res2.summary == "I'm not Ramon"
+        assert res2.data.texts["text"] == TextFieldData(
+            value=FieldText(
+                body="Really, I'm not Ramon",
+                format=TextFormat.PLAIN,
+                md5="62ef6847d003b438c597cb82d70982d0",
+            ),
+            extracted=None,
+            error=None,
+        )
+        assert res2.usermetadata == UserMetadata(
+            classifications=[
+                UserClassification(
+                    labelset="labelset", label="negative", cancelled_by_user=False
+                )
+            ],
+            relations=[],
+        )
+        assert res2.fieldmetadata == [
+            UserFieldMetadata(
+                token=[
+                    TokenSplit(
+                        token="Ferran",
+                        klass="NAME",
+                        start=5,
+                        end=9,
+                        cancelled_by_user=False,
+                    )
+                ],
+                paragraphs=[],
+                field=FieldID(field_type=FieldID.FieldType.TEXT, field="text"),
+            )
+        ]
+
+    # Validate upload updates fields
+    resource_id = knowledgebox.upload(
+        key="mykey1",
+        binary=File(data=b"asd", filename="data"),
+        text="I'm Ramon",
+        labels=["labelset/positive"],
+        entities=[Entity(type="NAME", value="Ramon", positions=[(5, 9)])],
+        vectors=[Vector(value=[1.0, 0.2], vectorset="base")],
+    )
+    resource1 = knowledgebox[resource_id]
+
+    knowledgebox.upload(
```

Comment:
Would be nice to have a method called `update` instead of reusing the `upload`

Reply:
Agreed. Done with `update_resource` and `create_resource`